Sources/AppcuesKit/Presentation/Traits/TraitMetadataDelegate.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  TraitMetadataDelegate.swift
+//  AppcuesKit
+//
+//  Created by Matt on 2022-12-06.
+//  Copyright © 2022 Appcues. All rights reserved.
+//
+
+import UIKit
+
+/// Methods for managing data to be shared across trait instances.
+@objc
+public class TraitMetadataDelegate: NSObject {
+    private var metadata: [String: Any?] = [:]
+    private var previousMetadata: [String: Any?] = [:]
+
+    private var subscribers: [String: (TraitMetadata) -> Void] = [:]
+    private var viewAnimatingSubscribers: [String: (TraitMetadata) -> Void] = [:]
```

Comment:
on first glance i thought something would get double published if it was in both, maybe could name `subscribers` to `nonAnimatingSubscribers` or something like that - to make it clear these are mutually exclusive sets.

Reply:
Good idea!